### PR TITLE
[configopaque, confmap] Change yaml test by confmap test

### PR DESCRIPTION
**Pull request by mx-psi**
**Description:** 

- Remove yaml test from configopaque
- Add test in confmap testing `configopaque`

**Link to tracking Issue:** From discussion on #9427

### CODE REVIEW

File: config/configopaque/opaque_test.go [hunk 1]
```diff
@@ -47,12 +46,6 @@ func TestStringJSON(t *testing.T) {
 	assert.Equal(t, `{"opaque":"[REDACTED]","plain":"plain"}`, string(bytes))
 }
 
-func TestStringYAML(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (evan-bradley) [reply config/configopaque/opaque_test.go:50]: Okay I see this justification now: https://github.com/open-telemetry/opentelemetry-collector/pull/9427#issuecomment-1917722325

I still think we should test that configopaque serializes correctly inside of configopaque and confmap can just use a dummy object to test that it works with the Go interfaces. But again I could have it backwards in my head.

**Comment by mx-psi** [comment]
Seems like you both agree on doing it the other way around, I prefer doing it this way but don't feel strongly about it, and we can always change it after the fact, so let me fix it :)

File: config/configopaque/opaque_test.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (mx-psi) [reply config/configopaque/opaque_test.go:50]: Fixed!

File: cmd/mdatagen/go.mod [hunk 2]
```diff
@@ -76,3 +76,5 @@ retract (
 	v0.76.1
 	v0.65.0
 )
+
+replace go.opentelemetry.io/collector/config/configopaque => ../../config/configopaque
```
[the remark this answers is not in the document]
Comment (mx-psi) [reply cmd/mdatagen/go.mod]: Changed

**Review by evan-bradley — APPROVED** [review]
Honestly the more I think of it, this really ought to be an integration test somewhere else in the repo. What we're really checking is that both the config packages and confmap use the same `encoding` interfaces when neither depends on the other.

That said, this looks good to me.

**Comment by mx-psi** [comment]
> Honestly the more I think of it, this really ought to be an integration test somewhere else in the repo. What we're really checking is that both the config packages and confmap use the same encoding interfaces when neither depends on the other.

That makes sense to me. I think for now we can go down this route since this is the status quo and changing this is not a breaking change

**Review by bogdandrutu — APPROVED** [review]

**Pull request merged**